### PR TITLE
Harden SelectMultiple against null value

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Prevent application crash when `value={null}` in SelectMultiple

#### What testing has been done on this PR?
Added jest test and changed existing stories to have `value={null}`

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [x] `asFragment()` is used for snapshot testing.

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No, component hasn't been released yet
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/SelectMultiple/SelectionSummary.js [hunk 1]
```diff
@@ -54,28 +56,30 @@ const SelectionSummary = ({
...
+            {value?.length === 0 || onMore || !value
+              ? `${value?.length || '0'} selected`
+              : `${value?.length || '0'} selected of ${options.length}`}
           </Text>
         </Box>
         {(options.length &&
-          (!limit || !(value.length === 0 && selectedValuesDisabled()))) > 0 &&
-          (!onMore || (onMore && value.length !== 0)) && (
+          (!limit ||
+            !(!value || (value?.length === 0 && selectedValuesDisabled())))) >
```
Comment (halocline) [inline src/js/components/SelectMultiple/SelectionSummary.js:66]: Lines 66, 71, 76, 82 all have the same condition and we are repeating ourselves. Can we assign the result of the condition to a variable and use the var instead?

**Review by halocline — COMMENTED** [review]
All the logic checks out and handles null/undefined value properly.

Made a couple comments around readability and consolidation.

File: src/js/components/SelectMultiple/SelectionSummary.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectionSummary.js:71]: Good suggestion
Added in latest commit

File: src/js/components/SelectMultiple/SelectMultipleContainer.js [hunk 2]
```diff
@@ -266,37 +268,39 @@ const SelectMultipleContainer = forwardRef(
...
+                result = disabledProp.indexOf(index) !== -1;
+              } else {
+                result =
+                  disabledProp.indexOf(
+                    getOptionValue(index, options, valueKey || labelKey),
+                  ) !== -1;
+              }
+            }
+            return result;
+          };
```
[the remark this answers is not in the document]
Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectMultipleContainer.js]: I agree, this helps with readability
Done in latest commit

File: src/js/components/SelectMultiple/SelectMultiple.js [hunk 3]
```diff
@@ -239,7 +239,7 @@ const SelectMultiple = forwardRef(
       if (valueLabel instanceof Function) {
```
[the remark this answers is not in the document]
Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectMultiple.js]: Added this in latest commit

**Review by halocline — APPROVED** [review]
Nice, thank you.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**